### DISCUSSION TITLE
[Question]: DALL-E/dall-e-3 Can AzureKey be used?

### DISCUSSION THREAD

**Discussion by 249805118** [danny-avila/LibreChat#1351] [Q&A]
### What is your question?

I don't have an openai account, I use AzureKey. Want to know current support?

### More Details

I don't have an openai account, I use AzureKey. Want to know current support?

### What is the main subject of your question?

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
You can set both a reverse proxy and specific api key for Dall-e. It may not work though, since azure is not usually 1:1 to OpenAI. I recently got Azure access so I can test it soon

**Reply by 249805118** [reply]
I tried, but it didn't work. Let me know what you find. Thank you

**Reply by danny-avila** [reply]
It needs to be developed, still have yet to start but should be simple when I do. Getting to this tomorrow

**Reply by danny-avila** [reply]
UPDATE: Answer is yes. Azure with DALL-E is working:

**For what it's worth, this is working for me:**
```.env
AZURE_OPENAI_MODELS=gpt-4-turbo
PLUGINS_USE_AZURE=true
AZURE_API_KEY=API-KEY-FOR-MY-INSTANCE
AZURE_USE_MODEL_AS_DEPLOYMENT_NAME=true
AZURE_OPENAI_API_INSTANCE_NAME=MY-INSTANCE
AZURE_OPENAI_API_DEPLOYMENT_NAME=gpt-35-turbo
AZURE_OPENAI_API_VERSION=2023-12-01-preview

DALLE3_AZURE_API_VERSION=2023-12-01-preview
DALLE3_BASEURL=https://MY-DALL-E-INSTANCE.openai.azure.com/openai/deployments/dall-e-3/
DALLE3_API_KEY=KEY-FOR-MY-DALL-E-INSTANCE
```
**gpt-4-turbo using Azure on Plugins endpoint**
![image](https://github.com/danny-avila/LibreChat/assets/110412045/7d76a878-f8ec-4369-a473-5919e98e3a16)

**Same model using Azure Endpoint**
![image](https://github.com/danny-avila/LibreChat/assets/110412045/64ff3d0a-6546-4d7b-b896-ebb805f9cdb8)


**For the examples above, API payload is identical for both (as expected when using `PLUGINS_USE_AZURE`)**

```js
{
  baseURL: "https://MY-INSTANCE.openai.azure.com/openai/deployments/gpt-4-turbo", // note the model is used as deployment name
  defaultQuery: {
    "api-version": "2023-12-01-preview", // from AZURE_OPENAI_API_DEPLOYMENT_NAME
  },
  defaultHeaders: {
    "api-key": "API-KEY-FOR-MY-INSTANCE",
  },
}
```

![image](https://github.com/danny-avila/LibreChat/assets/110412045/9eca2365-0bdf-4447-b7ec-38b2da37179e)

**Reply by TomGem** [reply]
Great, thank you! 

I tried this before, it didn't work. Turned out to be just one line, I had to change
`DALLE3_BASEURL=https://${AZURE_OPENAI_API_INSTANCE_NAME}.openai.azure.com/openai/deployments/dall-e-3/`
to
`DALLE3_BASEURL=https://actual-name-of-the-api-instance.openai.azure.com/openai/deployments/dall-e-3/`
in the `.env` file, although AZURE_OPENAI_API_INSTANCE_NAME is defined there.
